src/OrchardCore.Modules/OrchardCore.Lists/Models/ContainedPart.cs

```diff
@@ -12,6 +12,11 @@ public class ContainedPart : ContentPart
         /// </summary>
         public string ListContentItemId { get; set; }
 
+        /// <summary>
+        /// The content type of the list owning this content item.
+        /// </summary>
+        public string ListContentType { get; set; }
```

Comment:
@jtkech 
I am struggling to understand why do we need to to add `DisplayText`, `VersionOptions` as part of the index here? Also, why do we need the `IDX_ContainedPartIndex_DocumentId`?

Reply:
So because the `ContainedPart` has a new `ListContentType` field, good to update the related index table, so we can now have queries that filter on a container type, then good to also have the booleans `Published` and `Latest` (item states) to be able to filter on them too.

Look at the migrations of the `PublishLater` feature to see where we put these fields at the end of new index tables. Then the `DisplayText` as in `ContentItemIndex` is often useful, with it and an item Id we all have what we need to generate a link without fully loading a contentItem. Personally I put this field at the beginning after the identifier(s) columns.

> Also, why do we need the IDX_ContainedPartIndex_DocumentId?

It is already there and would need to be completed if we do the above. In fact what we call Index is a regular Table that can have inner sql index(s) that may include all or part of columns of its parent Table. It is used for perf, when a query uses/returns some columns it prevents some database hits if these columns are included in the inner index.

For example our joined queries on `ContentItemIndex` use the `DocumentId` to make the relation with the Document table, so in such queries we are sure that the `DocumentId` column will be always used. We are used to only have one inner index per Index Table (if possible) and that includes all columns of the related Table (if possible).

Finally when we do a direct `session.QueryIndex()` (not a joined query) to grab some data, the result contains all columns including the `Id` column, that's why I now also include it in new/updated `IDX_` inner index, as done in a migration step of the `PublishLater` feature.

But don't worry, will see, need first to fork and try your branch.